File: java/src/adapter/ObjectAdapterExample/VenderFactory.java
package adapter.ObjectAdapterExample;

public class VenderFactory {
		private static VenderFactory instance;
		
		private VenderFactory() {}
		
		public static VenderFactory getInstance() {
			if (instance == null) 
				instance = new VenderFactory();
			return instance;
		}
		
		public Vender createVender() {
			return new VenderAdapter( new Vender2());
		}

}

File: src/adapter/ObjectAdapterExample/VenderFactory.kt
package adapter.ObjectAdapterExample

class VenderFactory private constructor() {
    fun createVender(): Vender {
        return VenderAdapter(Vender2())
        //return ConcreteVender()
    }

    companion object {
        private var instance: VenderFactory? = null
        fun getInstance() : VenderFactory? {
            if (instance == null) instance = VenderFactory()
            return instance
        }
    }
}
